frame/staking/src/migrations.rs

```diff
@@ -287,217 +291,228 @@ pub mod v10 {
-// 	#[storage_alias]
-// 	type SnapshotNominators<T: Config> = StorageValue<Pallet<T>, ()>;
-// 	#[storage_alias]
-// 	type QueuedElected<T: Config> = StorageValue<Pallet<T>, ()>;
-// 	#[storage_alias]
-// 	type QueuedScore<T: Config> = StorageValue<Pallet<T>, ()>;
-// 	#[storage_alias]
-// 	type EraElectionStatus<T: Config> = StorageValue<Pallet<T>, ()>;
-// 	#[storage_alias]
-// 	type IsCurrentSessionFinal<T: Config> = StorageValue<Pallet<T>, ()>;
-//
-// 	/// check to execute prior to migration.
-// 	pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
-// 		// these may or may not exist.
-// 		log!(info, "SnapshotValidators.exits()? {:?}", SnapshotValidators::<T>::exists());
-// 		log!(info, "SnapshotNominators.exits()? {:?}", SnapshotNominators::<T>::exists());
-// 		log!(info, "QueuedElected.exits()? {:?}", QueuedElected::<T>::exists());
-// 		log!(info, "QueuedScore.exits()? {:?}", QueuedScore::<T>::exists());
-// 		// these must exist.
-// 		assert!(
-// 			IsCurrentSessionFinal::<T>::exists(),
-// 			"IsCurrentSessionFinal storage item not found!"
-// 		);
-// 		assert!(EraElectionStatus::<T>::exists(), "EraElectionStatus storage item not found!");
-// 		Ok(())
-// 	}
-//
-// 	/// Migrate storage to v6.
-// 	pub fn migrate<T: Config>() -> Weight {
-// 		log!(info, "Migrating staking to ObsoleteReleases::V6_0_0");
-//
-// 		SnapshotValidators::<T>::kill();
-// 		SnapshotNominators::<T>::kill();
-// 		QueuedElected::<T>::kill();
-// 		QueuedScore::<T>::kill();
-// 		EraElectionStatus::<T>::kill();
-// 		IsCurrentSessionFinal::<T>::kill();
-//
-// 		StorageVersion::<T>::put(ObsoleteReleases::V6_0_0);
-//
-// 		log!(info, "Done.");
-// 		T::DbWeight::get().writes(6 + 1)
-// 	}
-// }
+pub mod v9 {
+	use super::*;
+	#[cfg(feature = "try-runtime")]
+	use frame_support::codec::{Decode, Encode};
+	#[cfg(feature = "try-runtime")]
+	use sp_std::vec::Vec;
+	use frame_election_provider_support::{SortedListProvider, ReadOnlySortedListProvider};
+
+	pub trait MigrationConfig {
+		type Config: Config;
+		type VoterList: frame_election_provider_support::SortedListProvider<
+			<Self::Config as frame_system::Config>::AccountId,
+			Score = u64
+		>;
+	}
+
+	/// Migration implementation that injects all validators into sorted list.
+	///
+	/// This is only useful for chains that started their `VoterList` just based on nominators.
+	pub struct InjectValidatorsIntoVoterList<T>(sp_std::marker::PhantomData<T>);
+	impl<T: MigrationConfig> OnRuntimeUpgrade for InjectValidatorsIntoVoterList<T> {
+		fn on_runtime_upgrade() -> Weight {
+			if obsolete::StorageVersion::<T::Config>::get() == obsolete::Releases::V8_0_0 {
+				let prev_count = T::VoterList::count();
+				let weight_of_cached = Pallet::<T::Config>::weight_of_fn();
+				for (v, _) in Validators::<T::Config>::iter() {
+					let weight = weight_of_cached(&v);
+					let _ = T::VoterList::on_insert(v.clone(), weight).map_err(|err| {
```

Comment:
The migrations have been commented out for a reason. We don't have `on_insert` in `VoterList` in Staking anymore. Because it gets only read-only versions of bagsList.

Reply:
@kianenigma ^

Reply:
disregard